sql/core/src/main/scala/org/apache/spark/sql/execution/CollectMetricsExec.scala

```diff
@@ -60,16 +62,22 @@ case class CollectMetricsExec(
   override protected def doExecute(): RDD[InternalRow] = {
     val collector = accumulator
     collector.reset()
+    val updaterMap = mutable.Map[Long, AggregatingAccumulator]()
```

Comment:
Does this really work? This map is defined on the driver side and then we are updating it on the executor side.

Reply:
This works. The problem happens on executor side so the map also should be on the executor side.
But I've tried another approach by adding `addState`.